satpy/tests/test_readers.py

```diff
@@ -626,6 +627,48 @@ def test_old_reader_name_mapping(self):
             get_valid_reader_names([test_reader])
 
 
+def test_find_files_and_readers_fsspec_fsfile(tmp_path):
+    """Test that if an fsspec instance is passed, fsfile is returned."""
+    p = (tmp_path /
+         "OR_ABI-L1b-RadF-M3C01_G16_s19000010000000_e19000010001000"
+         "_c20152950029000.nc")
+    p.touch()
+
+    from satpy.readers import FSFile, find_files_and_readers
+
+    ri = find_files_and_readers(
+            base_dir=p.parent,
+            fs=fsspec.implementations.local.LocalFileSystem(),
+            reader="abi_l1b")
+
+    assert ri.keys() == {"abi_l1b"}
+    assert len(ri["abi_l1b"]) == 1
+    assert isinstance(ri["abi_l1b"][0], FSFile)
+    assert os.path.normpath(str(ri["abi_l1b"][0])) == os.path.normpath(str(p))
+
+
+def test_find_files_and_readers_fsspec_not_available(tmp_path):
+    """Test that if fsspec is not installed filepath is returned as a string."""
+    p = (tmp_path /
+         "OR_ABI-L1b-RadF-M3C01_G16_s19000010000000_e19000010001000"
+         "_c20152950029000.nc")
+    p.touch()
+
+    with mock.patch.dict('sys.modules', {'fsspec': None}):
+        # delete find_files_and_readers import and reload
+        del sys.modules['satpy.readers']
```

Comment:
Maybe it's this line that causes all the failures?

Reply:
That looks like a candidate for the [`monkeypatch`](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html) (`with monkeypatch.delitem(sys.modules, "satpy.readers"): ...`

Reply:
@pnuu yepp that's it. The goal here is to really test the try/except import of fsspec. But apparently the deletion of `satpy.readers` is global and not restricted to the `with` block. It also seems like the patching does not work as I expected. In tried to have a look how this is done elsewhere in Satpy and while there are some try/except imports it seems that this is actually tested nowhere?!
@gerritholl I have not tried with the monkeypatch yet.